docs/solution-architecture/smartDocumentsIntegration.md

```diff
@@ -1,97 +1,54 @@
-# SmartDocuments document creation flow
+# SmartDocuments
+
+## Configuration 
+
+The configuration of every case type contains a mapping between SmartDocuments template and document type. This mapping is used to get the correct template and template group with the Document creation flow below.
+
+## Document creation flow
 
 The ZAC - SmartDocuments document creation flow is as follows:
 
 1. User starts SmartDocument wizard from ZAC ('create document'). This can be done either from a case (zaak) or from a task.
 2. ZAC sends a HTTPS request to start a document creation wizard to SmartDocuments and sends case- and other information in this request.
-3. The document creation wizard is started in SmartDocuments. Case or task information is pre-filled.
-4. When the user closes the document creation wizard, SmartDocuments does the following by sending the following HTTPS requests using the SmartDocuments 'DMS Plugin' (see below):
-    1. A Word document ('enkelvoudiginformatieobject') is created in the ZRC API implementation (also known as 'zaakregister'; in the context of ZAC this is usually OpenZaak).
-    2. Usage permissions ('Gebruiksrechten') are set for the document. Without this step the zaak to which this document belongs cannot be closed nor interrupted.  
-    3. The document/enkelvoudiginformatieobject is linked to the case or task in question (by creating a ‘zaakinformatieobject’).
-5. The zaakregister sends a notification to Open Notificaties that a new document has been linked to a case or task.
-6. Open Notificaties sends a callback request to ZAC informing ZAC of this event.
-7. Depending on which SmartDocuments environment is used and how it is configured, the user also has the option to download the document to their local computer.
-8. When the user returns to the ZAC browser tab, they will see the new document attached to the case or task.
+3. The document creation wizard is started in SmartDocuments. Case or task information and template type are pre-filled.
+4. At the end of the document creation wizard, SmartDocuments:
+   1. creates a Word document and stores it on SmartDocuments side for later retrieval
+   2. calls a ZAC callback endpoint with information about the created document
+5. ZAC callback endpoint in turn:
+   1. Downloads the Word document from SmartDocuments 
+   2. Creates a document ('enkelvoudiginformatieobject') in the ZRC API implementation (also known as 'zaakregister'; in the context of ZAC this is usually OpenZaak).
+   3. Usage permissions ('Gebruiksrechten') are set for the document. Without this step the zaak to which this document belongs cannot be closed nor interrupted.  
+   4. The document/enkelvoudiginformatieobject is linked to the case or task in question (by creating a ‘zaakinformatieobject’).
```

Comment:
Add something on 'confidentiality' level applied to the document?

Reply:
Added:
`3. Confidentiality of the document is set to PUBLIC and the document status to IN_PROGRESS`